cache.go

```diff
@@ -1,5 +1,17 @@
 package main
 
+/**
```

Comment:
can you move the comments above the package declaration? It will read better in godoc